harvester/files/tests/factories/youtube.py

```diff
@@ -0,0 +1,104 @@
+import json
+from urllib.parse import quote
+import factory
+
+from files.models import YoutubeAPIResource
+
+
+class HttpYoutubeResourceFactory(factory.django.DjangoModelFactory):
+
+    class Meta:
+        model = YoutubeAPIResource
+
+    class Params:
+        id = ""
+        type = "videos"
+
+    status = 200
+    data_hash = ""
+
+    @factory.lazy_attribute
+    def uri(self):
+        id = quote(self.id, safe="")
+        type = quote(self.type, safe="videos")
+        return f"youtube.googleapis.com/youtube/v3/{type}?id={id}&part=snippet%2Cplayer%2CcontentDetails%2Cstatus"
+
+    @factory.lazy_attribute
+    def request(self):
+        url = "http://" + self.uri
+        return {
+            "url": url,
+            "args": [self.id],
```

Comment:
Type is also given as an args to YoutubeAPIResource.get, right? So that should be part of the args list here.